webapp/api/requests.py

```diff
@@ -0,0 +1,138 @@
+# Core packages
+import os
+from urllib.parse import urlparse
+
+# Third-party packages
+import requests as _requests
+import requests_cache
+import prometheus_client
+
+# Local packages
+from webapp.api.exceptions import (
+    ApiTimeoutError,
+    ApiConnectionError
+)
+
+
+timeout_counter = prometheus_client.Counter(
+    'feed_timeouts',
+    'A counter of timed out requests',
+    ['domain'],
+)
+connection_failed_counter = prometheus_client.Counter(
+    'feed_connection_failures',
+    'A counter of requests which failed to connect',
+    ['domain'],
+)
+latency_histogram = prometheus_client.Histogram(
+    'feed_latency_seconds',
+    'Feed requests retrieved',
+    ['domain', 'code'],
+    buckets=[0.25, 0.5, 0.75, 1, 2, 3],
+)
+
+
+class TimeoutHTTPAdapter(_requests.adapters.HTTPAdapter):
+    def __init__(self, timeout=None, *args, **kwargs):
+        self.timeout = timeout
```

Comment:
This seems insanely complex, but I agree there doesn't seem to be another way. *sigh*